### PR TITLE
python3Packages.flask-gravatar: init at 0.5.0

### PR DESCRIPTION
**Pull request by gador**
Init of Flask-Gravatar

###### Motivation for this change
needed for pgadmin4 init
https://github.com/NixOS/nixpkgs/pull/154764

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) 
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/flask-gravatar/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    pytest-cov
+    pygments
+  ];
+
+  # use pycodestyle as a replacement for the removed pep8.
+  # Since there is no pytest-pycodestyle plugin in nixpkgs
+  # use it separately after the pytestCheckHook
+  postCheck = ''
+    ${pycodestyle}/bin/pycodestyle --statistics --ignore=E305,E501,W293 \
+                                   --exclude=docs/conf.py,docs/_themes/flask_theme_support.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-gravatar/default.nix:50]: Does this only check for style things or has real tests? if it is only the former we can skip it.

File: pkgs/development/python-modules/flask-gravatar/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  # pep8 is deprecated
+  preCheck = ''
+    substituteInPlace pytest.ini --replace "--pep8" ""
+  '';
+
+  checkInputs = [
+    pytestCheckHook
+    pycodestyle
+    pytest-cov
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-gravatar/default.nix:41]:
```suggestion
```
no reason to collect coverage.

File: pkgs/development/python-modules/flask-gravatar/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+     -e "s|extras_require=extras_require,||g" \
+     -e "s|setup_requires=setup_requires,||g"
+  '';
+
+  propagatedBuildInputs = [
+    flask
+  ];
+
+  # pep8 is deprecated
+  preCheck = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-gravatar/default.nix:34]: That also goes into postPatch.

File: pkgs/development/python-modules/flask-gravatar/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    sha256 = "YGZfMcLGEokdto/4Aek+06CIHGyOw0arxk0qmSP1YuE=";
+  };
+
+
+  preBuild = ''
+    sed -i setup.py \
+     -e "s|tests_require=tests_require,||g" \
+     -e "s|extras_require=extras_require,||g" \
+     -e "s|setup_requires=setup_requires,||g"
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-gravatar/default.nix:27]: That should be postPatch.

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -4250,6 +4250,16 @@
...
+  gador = {
+    email = "florian.brandes@posteo.de";
+    github = "gador";
+    githubId = 1883533;
+    name = "Florian Brandes";
+    keys = [{
+      longkeyid = "rsa4096/0xBBB3E40E53797FD9";
+      fingerprint = "0200 3EF8 8D2B CF2D 8F00  FFDC BBB3 E40E 5379 7FD9";
+    }];
+  };
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4262]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The init commit should contain the version.

File: pkgs/development/python-modules/flask-gravatar/default.nix (hunk 1 above)
Comment (gador) [reply pkgs/development/python-modules/flask-gravatar/default.nix:50]: the original tests do unittests and also a pep8 check. I replaced the pep8 check with the newer pycodestyle. 
Sure, I''ll remove it

**Comment by gador** [comment]
@ofborg eval

File: pkgs/development/python-modules/flask-gravatar/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  version = "0.5.0";
+
+  src = fetchPypi {
+    pname = "Flask-Gravatar";
+    inherit version;
+    sha256 = "YGZfMcLGEokdto/4Aek+06CIHGyOw0arxk0qmSP1YuE=";
+  };
+
+
+  postPatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-gravatar/default.nix:20]:
```suggestion
  };

  postPatch = ''
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 155214` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python310Packages.flask-gravatar</li>
<li>python39Packages.flask-gravatar</li>
</ul>
</details>
